examples/fabric-sync/shell/ShellCommands.cpp

```diff
@@ -197,6 +229,10 @@ static CHIP_ERROR AppPlatformHandler(int argc, char ** argv)
     {
         return HandleAddDeviceCommand(argc, argv);
     }
+    else if (strcmp(argv[0], "pair-device") == 0)
```

Comment:
Generally speaking why are we adding a new command when you can already do `pairing onnetwork ...`? Knowing the answer to this question would really help me review this PR faster as I will know what difference I am trying to look for

Reply:
We don't have 'pairing onnetwork' command in fabric-sync since it is not inherited from chip-tool, basically we need to implement all commands we need to do fabric-sync.

This command allow to pair a simulated matter device started on the same machine, then we don't need to use a separate matter device to test fabric-sync, just like what we did in the cert test.